contrib/craftscripts/maze.js

```diff
@@ -21,363 +21,369 @@ importPackage(Packages.com.sk89q.worldedit);
+help += "§7• §6r §r: Places red wool if wrong\n";
+help += "§7• §6b §r: Places blue wool if unvisited";
 
 sess = context.remember();
 origin = player.getBlockIn().toVector().toBlockPoint();
 
 // This may throw an exception that is caught by the script processor
-block = context.getBlock(argv[1]);
-airBlock = context.getBlock("air");
-glassBlock = context.getBlock("glass");
-limeWoolBlock = context.getBlock("lime_wool");
-redWoolBlock = context.getBlock("red_wool");
-blueWoolBlock = context.getBlock("blue_wool");
-
-if (argv.length > 7) flags = String(argv[7]);
-else flags = false;
-
-if (argv.length > 6) {
-    if (parseInt(argv[6], 10)) wa = parseInt(argv[6], 10);
-    else flags = String(argv[6]), wa = 1;
-} else wa = 1;
-
-if (argv.length > 5) {
-    if (parseInt(argv[5], 10)) s = parseInt(argv[5], 10);
-    else flags = String(argv[5]), s = 1, wa = 1;
-} else s = 1;
-
-if (argv.length > 4) {
-    if (parseInt(argv[4], 10)) h = parseInt(argv[4], 10);
-    else flags = String(argv[4]), h = 2, s = 1, wa = 1;
-} else h = 2;
-
-if (argv.length > 3) {
-    if (parseInt(argv[3], 10)) l = parseInt(argv[3], 10);
-    else flags = String(argv[3]), l = 5, h = 2, s = 1, wa = 1;
-} else l = 5;
-
-if (argv.length > 2) {
-    if (parseInt(argv[2], 10)) w = parseInt(argv[2], 10);
-    else flags = String(argv[2]), w = 5, l = 5, h = 2, s = 1, wa = 1;
-} else w = 5;
-
-if (flags) {
-    ee = flags.search("i") != -1;
-    r = flags.search("y") != -1;
-    if (r) ee = true;
-    f = flags.search("f") != -1;
-    c = flags.search("c") != -1;
-    e = flags.search("e") != -1;
-    ao = flags.search("a") != -1;
-    if (ao) f = c = false, e = true;
-    v = flags.search("v") != -1;
-    so = flags.search("s") != -1;
-    if (so) ee = true;
-    g = flags.search("g") != -1;
-    re = flags.search("r") != -1;
-    bl = flags.search("b") != -1;
-    if (g || re || bl) so = ee = true;
-} else ee = r = f = c = e = ao = v = so = g = re = bl = false;
-
-function id(x, y) {
-    return y * (w + 1) + x;
-}
-
-function $x(i) {
-    return i % (w + 1);
-}
-
-function $y(i) {
-    return Math.floor(i / (w + 1));
-}
-
-function shuffle(arr) {
-    i = arr.length;
-    if (i === 0) return false;
-    while (--i) {
-        j = Math.floor(Math.random() * (i + 1));
-        tempi = arr[i];
-        tempj = arr[j];
-        arr[i] = tempj;
-        arr[j] = tempi;
-    }
-}
-
-if (f || c) {
-    for (z = 1; z <= wa; z++) for (y = -wa; y <= l * (s + wa) - 1; y++) for (x = -wa; x <= w * (s + wa) - 1; x++) {
-        if (f) {
-            if (!v) sess.setBlock(origin.add(x, -z, y), block);
-            else sess.setBlock(origin.add(x, y, +z), block);
-        }
-
-        if (c) {
-            if (!v) sess.setBlock(origin.add(x, z - 1 + h, y), block);
-            else sess.setBlock(origin.add(x, y, -z + 1 - h), block);
-        }
+if (!argv[1]) context.print(usage);
+else if (argv[1] == "?") context.print(help);
+else {
+    block = context.getBlock(argv[1]);
```

Comment:
Rather than dump the entire rest of the script into an `else`, would it be possible to throw an exception / error after the `context.print`? I feel like it may not be super user-friendly, and we should probably add some sort of `exit(Component)`, but having literally the entire rest of the script indented is a big 👎 from me.